### PR TITLE
chore: fix dependabot alerts

### PR DESCRIPTION
**Pull request by rchincha**
**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by rchincha** [comment]
https://github.com/opencontainers/image-spec/commit/25fc5534d6a7a206f993ba03d63199eea22d7136#diff-f644ae378dd9d27890b86e85b2427c53f10853e498ee3aa38c794b28b879c9d8

^ has been merged

File: go.mod [hunk 1]
```diff
@@ -473,3 +483,5 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
+
+replace github.com/google/go-containerregistry => github.com/google/go-containerregistry v0.14.1-0.20230409045903-ed5c185df419
```
Comment (andaaron) [inline go.mod:487]: Why upgrade this package and then use a replace here for an older version?

Comment (rchincha) [reply go.mod:487]: because trivy and cosign deps are for different versions and the newer version breaks.

Comment (andaaron) [reply go.mod:487]: Maybe it's better to use top of main for cosign?

**Review by andaaron — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (andaaron) [reply go.mod:487]: Actually I think we're ok if everything compiles successfully and the tests pass

**Pull request merged**